### ISSUE TITLE
[Bug] Cannot create a file when that file already exists

### ISSUE THREAD

**Issue by bigboss97** [deezer/spleeter#163]
## Description
I use the option "-o" to specify the output directory. It automatically creates sub-directory with the song title, e.g.
allSongs/Happy Birthday
Beneath that 2 files are created: accompaniment.wav and vocals.wav
For the 2nd song I use the same director and it creates:
allSongs/Jingle Bells/accompaniment.wav
Then it stops with:
FileExistsError: [Errno 17] Cannot create a file when that file already exists:

Jingle Bells/accompaniment.wav was created successfully.

## Step to reproduce
1. Run spleeter to convert song1 with -o allSongs.
2. Run spleeter to convert song2 with -o allSongs.
song2/accompaniment.wav created and fails to create vocals.wav.

## Output
FileExistsError: [Errno 17] Cannot create a file when that file already exists: song2

## Environment
Windows 7

## Additional context
Workaround:
1. Rename song1/vocals.wav to vocals1.wav
2. Delete song2 directory.
3. Rerun spleeter for song2.

**Comment by ngoname** [comment]
This doesn't work all of the time.

**Comment by bigboss97** [comment]
Just had the opposite case, vocals was generated and had to rename accompaniment of previous song to workaround.

**Comment by webbnel** [comment]
Try making a new folder inside the folder you are working named spleeter and put the song in there. The full command line should look like this:

 C:\Users\YourUsername\FolderOfYourMusic>python -m spleeter separate -i spleeter/YourSong.mp3 -p spleeter:2stems -o OutputFolderName

You may have to change folder location in cmd by using cd.

**Comment by the-pont** [comment]
I'm using spleeter version 1.4.9 on Windows 10 (haven't tried on another OS yet), and was running into this same problem. Not every time, but most of the time. I made one change to the source code, and after that things worked for me.

in audio/ffmpeg.py I changed line 108 from
`os.makedirs(directory)`
to this
`os.makedirs(directory, exist_ok=True)`

Looks like this project is using multiprocessing to create the wav files, and in that case, the check performed on line 107  `if not os.path.exists(directory):` will not guarantee that the directory is not created, because another process could have created the directory already. In my opinion the check could be removed completely, since the exist_ok option does what that check intends anyway.

When I get a chance to test more thoroughly, I can confirm if this completely fixed the issue or not.
